### DISCUSSION TITLE
how to decrypt or encrypt a user password with command lines

### DISCUSSION THREAD

**Discussion by shakibamoshiri** [SoftEtherVPN/SoftEtherVPN#1778] [General]
if a user password be forgotten, we can get that user info with RPC-JSON. but it is encrypted.
How to decrypt with Linux commands and encrypt again
Has anyone done this?

```json
    "GroupName_str": "",
    "HashedKey_bin": "xxxxxxxxxxxxxxxxxxxxxx=",
    "HubName_str": "xxx",
    "Name_str": "xxxxx",
    "Note_utf": "xxxx",
    "NtLmSecureHash_bin": "xxxxxxxxxxxxxxxxxxxxxxxxxx==",
    "NumLogin_u32": 0,
```

**Comment by domosekai** [comment]
You can't because password is hashed.

**Comment by ccalvocm** [comment]
Greetings @shakibamoshiri :

Let's assume that json is saved as a file named jsonTest.json.
1. Using Linux command 'jq' you can get the encrypted hash from your .json file and output it to a plain text file using this Linux command:

`jq -r '.NtLmSecureHash_bin' jsonTest.json > hash.hash`

3. Install  hashcat:

`sudo apt install hashcat-nvidia`

4. decrypt your hash.hash file using this command, asumming the encryption algorithm is NTLM from your json file:

`hashcat -m 1000 hash.hash -o dec.hash`

5.  To ecrypt again you would run this on python:

`python
hashDec=open('dec.hash','r').read().strip()
hash = hashlib.new('ntlm', hashDec.encode('utf-16le')).digest()
#open text file
text_file = open("hash.hash", "w")
 
#write string to file
text_file.write(hash)
 
#close file
text_file.close()
